### PR TITLE
Use new HdSceneIndex APIs.

### PR DESCRIPTION
**Pull request by dj-mcg**
This should be a no-op until USD 23.11

### CODE REVIEW

**Comment by dj-mcg** [comment]
Hey folks! Looks like a test is failing for Maya 2022 Windows Py3 but I'm not sure how that's related to this change. Can you have a peek to let me know if there's anything I need to do on my end? Thanks!

**Comment by seando-adsk** [comment]
@dj-mcg It appears to just be a testing timeout glitch (happens sometimes). I'll have the code reviewed and then merge it.

File: lib/mayaUsd/sceneIndex/proxyShapeSceneIndexPlugin.cpp [hunk 1]
```diff
@@ -153,21 +170,30 @@ HdSceneIndexBaseRefPtr MayaUsdProxyShapeMayaNodeSceneIndexPlugin::_AppendSceneIn
             // Flatten transforms, visibility, purpose, model, and material
             // bindings over hierarchies.
             // Do it the same way as USDView does with a SceneIndices chain
-            HdSceneIndexBaseRefPtr _sceneIndex
+            HdSceneIndexBaseRefPtr sceneIndex
                 = UsdImagingPiPrototypePropagatingSceneIndex::New(usdImagingStageSceneIndex);
-            _sceneIndex = UsdImagingNiPrototypePropagatingSceneIndex::New(_sceneIndex);
+#if PXR_VERSION < 2311
```
Comment (lanierd-adsk) [inline lib/mayaUsd/sceneIndex/proxyShapeSceneIndexPlugin.cpp:175]: @seando-adsk : there is a missing case here which is for usd 23.08, it should contain : 
```
            //#For USD 23.08 and later, HD_API_VERSION=54 in USD 23.08
            static const bool _displayUnloadedPrimsWithBounds = true;

            HdContainerDataSourceHandle const stageInputArgs =
            HdRetainedContainerDataSource::New(
                UsdImagingStageSceneIndexTokens->includeUnloadedPrims,
                HdRetainedTypedSampledDataSource<bool>::New(
                    _displayUnloadedPrimsWithBounds));

            // Create the scene index graph.
            _sceneIndex = _stageSceneIndex =
                UsdImagingStageSceneIndex::New(stageInputArgs);

            static HdContainerDataSourceHandle const materialPruningInputArgs =
                HdRetainedContainerDataSource::New(
                    HdsiPrimTypePruningSceneIndexTokens->primTypes,
                    HdRetainedTypedSampledDataSource<TfTokenVector>::New(
                        { HdPrimTypeTokens->material }),
                    HdsiPrimTypePruningSceneIndexTokens->bindingToken,
                    HdRetainedTypedSampledDataSource<TfToken>::New(
                        HdMaterialBindingsSchema::GetSchemaToken()));

            // Prune scene materials prior to flattening inherited
            // materials bindings and resolving material bindings
            _sceneIndex = _materialPruningSceneIndex =
                HdsiPrimTypePruningSceneIndex::New(
                    _sceneIndex, materialPruningInputArgs);

            static HdContainerDataSourceHandle const lightPruningInputArgs =
                HdRetainedContainerDataSource::New(
                    HdsiPrimTypePruningSceneIndexTokens->primTypes,
                    HdRetainedTypedSampledDataSource<TfTokenVector>::New(
                        HdLightPrimTypeTokens()),
                    HdsiPrimTypePruningSceneIndexTokens->doNotPruneNonPrimPaths,
                    HdRetainedTypedSampledDataSource<bool>::New(
                        false));

            _sceneIndex = _lightPruningSceneIndex =
                HdsiPrimTypePruningSceneIndex::New(
                    _sceneIndex, lightPruningInputArgs);

            // Use extentsHint for default_/geometry purpose
            HdContainerDataSourceHandle const extentInputArgs =
                HdRetainedContainerDataSource::New(
                    UsdGeomTokens->purpose,
                    HdRetainedTypedSampledDataSource<TfToken>::New(
                        UsdGeomTokens->default_));

            _sceneIndex =
                UsdImagingExtentResolvingSceneIndex::New(
                    _sceneIndex, extentInputArgs);

            if (_displayUnloadedPrimsWithBounds) {
                _sceneIndex =
                    UsdImagingUnloadedDrawModeSceneIndex::New(_sceneIndex);
            }

            _sceneIndex = _rootOverridesSceneIndex =
                UsdImagingRootOverridesSceneIndex::New(_sceneIndex);

            _sceneIndex =
                UsdImagingPiPrototypePropagatingSceneIndex::New(_sceneIndex);
        
            _sceneIndex =
                UsdImagingNiPrototypePropagatingSceneIndex::New(_sceneIndex);

            _sceneIndex = _selectionSceneIndex =
                UsdImagingSelectionSceneIndex::New(_sceneIndex);

            _sceneIndex =
                UsdImagingRenderSettingsFlatteningSceneIndex::New(_sceneIndex);

            _sceneIndex =
                HdFlatteningSceneIndex::New(
                    _sceneIndex, UsdImagingFlattenedDataSourceProviders());

            _sceneIndex =
                UsdImagingDrawModeSceneIndex::New(_sceneIndex,
                                                  /* inputArgs = */ nullptr);

            _sceneIndex = _displayStyleSceneIndex =
                HdsiLegacyDisplayStyleOverrideSceneIndex::New(_sceneIndex);
```

**Review by lanierd-adsk — CHANGES_REQUESTED** [review]

File: lib/mayaUsd/sceneIndex/proxyShapeSceneIndexPlugin.cpp (hunk 1 above)
Comment (seando-adsk) [reply lib/mayaUsd/sceneIndex/proxyShapeSceneIndexPlugin.cpp:175]: @dj-mcg can you have a look at the code review comment here.

Comment (seando-adsk) [reply lib/mayaUsd/sceneIndex/proxyShapeSceneIndexPlugin.cpp:175]: @dj-mcg Dan would you mind if we pushed the change to the missing 23.08 case to this PR? We are using 23.08 internally and need this fix for MayaHydra.

File: lib/mayaUsd/sceneIndex/proxyShapeSceneIndexPlugin.cpp [hunk 2]
```diff
@@ -153,21 +170,30 @@ HdSceneIndexBaseRefPtr MayaUsdProxyShapeMayaNodeSceneIndexPlugin::_AppendSceneIn
...
+            sceneIndex = UsdImagingNiPrototypePropagatingSceneIndex::New(sceneIndex);
+#else
+            sceneIndex = UsdImagingNiPrototypePropagatingSceneIndex::New(
```
Comment (lanierd-adsk) [inline lib/mayaUsd/sceneIndex/proxyShapeSceneIndexPlugin.cpp:178]: If I am correct, this block will never be called as line 137 we already returned from the function for USD 23.11+.

**Comment by lanierd-adsk** [comment]
@seando-adsk  : I created a new PR https://github.com/Autodesk/maya-usd/pull/3293 which contains the changes from this PR and the fix for USD 23.08 so this PR could be closed.
@dj-mcg thank you for your contribution, it's been appreciated !

File: lib/mayaUsd/sceneIndex/proxyShapeSceneIndexPlugin.cpp (hunk 1 above)
Comment (dj-mcg) [reply lib/mayaUsd/sceneIndex/proxyShapeSceneIndexPlugin.cpp:175]: Oh, so sorry to have dropped the ball on this, I was OOO and am just catching up on things.

No problem checking this in, thanks for taking care of this case!

**Pull request closed without merging**